### ISSUE TITLE
unsolicited suggestion to simplify

### ISSUE THREAD

**Issue by imranariffin** [hbina/CapstoneAndroidApp#1]
@hbina @akiraro 

Sorry for busting in with unsolicited suggestion but I was reading the code and saw the TODO comment to simplify the code. I figured I might be able to provide some general suggestions.

Change from

https://github.com/hbina/CapstoneAndroidApp/blob/e7bf5354a9832a2829cc751ae971879f0016f792/app/src/main/java/com/smartstick/ceg4912/capstoneandroidapp/services/RfidServices.java#L36-L80

To

```
@Override
public void run() {
    super.run();
    // TODO: Rewrite this...make it a lot simpler than this...
    while (isRunning.get()) {
        try {
            if (BluetoothConnector.getInputStream() == null) {
                continue;
            }
            
            int availableBytes = BluetoothConnector.getInputStream().available();

            if (availableBytes <= 0) {
                continue;
            }

            Log.d(TAG, String.format("Received something from Bluetooth of size:%d", availableBytes));
            byte[] rawBytes = new byte[availableBytes];
            final int receivedLength = BluetoothConnector.getInputStream().read(rawBytes);
            final String receivedString = filterBluetooth((new String(rawBytes, StandardCharsets.UTF_8)));

            if (receivedLength <= 1) {
                Log.d(TAG, "Delta string recevied => " + receivedString);
                continue;
            }

            Log.d(TAG, "receivedString:" + receivedString);
            currentLocation = receivedString;

            if (nodesInPath.isEmpty()) {
                SpeechServices.addText("Please enter a destination...");
                callerActivity.TEXT_VIEW_PATH.setText(nodesInPath.toString());
                continue;
            }
            
            if (nodesInPath.size() == 1) {
                SpeechServices.addText("Congratulation, you have arrived at " + nodesInPath.poll());
                callerActivity.TEXT_VIEW_PATH.setText(nodesInPath.toString());
                continue;
            }
            
            if (!currentLocation.equals(decodeNodeNameToId(nodesInPath.peekFirst()))) {
                Log.d(TAG, String.format("currentLocation:%s peekFirst:%s", currentLocation, nodesInPath.peekFirst()));
                continue;
            }
            
            SpeechServices.addText("You have arrived at " + encodeIdToName(receivedString));
            nodesInPath.removeFirst();
            BearingRequest bearingRequest = new BearingRequest(currentLocation, nodesInPath.peekFirst(), BearingListener.getBearing());
            RequestServices.addBearingRequest(bearingRequest);

            callerActivity.TEXT_VIEW_PATH.setText(nodesInPath.toString());

        } catch (
                IOException exception) {
            Log.e(TAG, exception.getMessage());
        }
    }
    Log.d(TAG, "Thread have finished running");
}
```

The logic is exactly the same but the idea here is to reorganize the main branch to as left as possible to avoid nested conditionals. Nested conditionals are generally more difficult to understand.

Hope this is useful.

**Comment by imranariffin** [comment]
It's also generally good to maintain line length <= 80 chars so other devs wouldn't have to scroll horizontally on editors or github code review.

**Comment by hbina** [comment]
Thank you for the suggestion, this has been resolved with #2
